From patchwork Fri Jan 28 13:09:32 2022
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Michel Lespinasse <michel@lespinasse.org>
X-Patchwork-Id: 12728502
Return-Path: <owner-linux-mm@kvack.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from kanga.kvack.org (kanga.kvack.org [205.233.56.17])
	by smtp.lore.kernel.org (Postfix) with ESMTP id 0C550C433F5
	for <linux-mm@archiver.kernel.org>; Fri, 28 Jan 2022 13:10:11 +0000 (UTC)
Received: by kanga.kvack.org (Postfix)
	id 442056B0072; Fri, 28 Jan 2022 08:10:10 -0500 (EST)
Received: by kanga.kvack.org (Postfix, from userid 40)
	id 3D0286B0071; Fri, 28 Jan 2022 08:10:10 -0500 (EST)
X-Delivered-To: int-list-linux-mm@kvack.org
Received: by kanga.kvack.org (Postfix, from userid 63042)
	id 0EA006B007D; Fri, 28 Jan 2022 08:10:10 -0500 (EST)
X-Delivered-To: linux-mm@kvack.org
Received: from forelay.hostedemail.com (smtprelay0080.hostedemail.com
 [216.40.44.80])
	by kanga.kvack.org (Postfix) with ESMTP id ED1DF6B0071
	for <linux-mm@kvack.org>; Fri, 28 Jan 2022 08:10:09 -0500 (EST)
Received: from smtpin17.hostedemail.com (10.5.19.251.rfc1918.com
 [10.5.19.251])
	by forelay04.hostedemail.com (Postfix) with ESMTP id 9C8E494FD5
	for <linux-mm@kvack.org>; Fri, 28 Jan 2022 13:10:09 +0000 (UTC)
X-FDA: 79079728938.17.0B2446C
Received: from server.lespinasse.org (server.lespinasse.org [63.205.204.226])
	by imf16.hostedemail.com (Postfix) with ESMTP id 2D869180002
	for <linux-mm@kvack.org>; Fri, 28 Jan 2022 13:10:09 +0000 (UTC)
DKIM-Signature: v=1; a=ed25519-sha256; c=relaxed/relaxed;
 d=lespinasse.org; i=@lespinasse.org; q=dns/txt; s=srv-52-ed;
 t=1643375406; h=from : to : cc : subject : date : message-id :
 in-reply-to : references : mime-version : content-transfer-encoding :
 from; bh=60KVsRSybg58S5Bq7/6VWRR0Ob54Rjhy19T2JOiOHJE=;
 b=M7pzfh5ZrllwylgwXDFgMgUOxk+x0CQXf6B5IVcHP6kNbiwFEwwKpkHsAzgcdMs6ciQ+K
 40+M0hVJhJPDMYlCA==
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=lespinasse.org;
 i=@lespinasse.org; q=dns/txt; s=srv-52-rsa; t=1643375406; h=from : to
 : cc : subject : date : message-id : in-reply-to : references :
 mime-version : content-transfer-encoding : from;
 bh=60KVsRSybg58S5Bq7/6VWRR0Ob54Rjhy19T2JOiOHJE=;
 b=hH+5ulWntN/qZ08GjZofRdKqO2OeOaAGUxe42j5ciL7Bjfd4AWn2kXvECpD0Tk1kLuQim
 pXNlIDBs0nLbih3vTu1Xmftjg+LmMrBFYRCGmnBJXw+5vZdhI3IEbAvdi8t/VIWjDGOIWvb
 VK/0trH9WTTtLQsjBsY0lG/KIFr00ZEXA0HIUrpP7HHkEFIFaJjWppZIzD16uxbHJw/3rti
 LymDIcnLEYmmKHELb7Bb3LM2Q3tDBAZ1TqK6ebvJCW9rjIgkV5aWyrvSy/xo6LiJO1+KsYu
 PgxAcY2i2izS6gADabfEHywhVuqppThEPWfhOrm+GR6Sg4NMiy43Uk41aLmA==
Received: from zeus.lespinasse.org (zeus.lespinasse.org [10.0.0.150])
	by server.lespinasse.org (Postfix) with ESMTPS id CEF33160942;
	Fri, 28 Jan 2022 05:10:06 -0800 (PST)
Received: by zeus.lespinasse.org (Postfix, from userid 1000)
	id C2D5D1FF74; Fri, 28 Jan 2022 05:10:06 -0800 (PST)
From: Michel Lespinasse <michel@lespinasse.org>
To: Linux-MM <linux-mm@kvack.org>,
	linux-kernel@vger.kernel.org,
	Andrew Morton <akpm@linux-foundation.org>
Cc: kernel-team@fb.com,
	Laurent Dufour <ldufour@linux.ibm.com>,
	Jerome Glisse <jglisse@google.com>,
	Peter Zijlstra <peterz@infradead.org>,
	Michal Hocko <mhocko@suse.com>,
	Vlastimil Babka <vbabka@suse.cz>,
	Davidlohr Bueso <dave@stgolabs.net>,
	Matthew Wilcox <willy@infradead.org>,
	Liam Howlett <liam.howlett@oracle.com>,
	Rik van Riel <riel@surriel.com>,
	Paul McKenney <paulmck@kernel.org>,
	Song Liu <songliubraving@fb.com>,
	Suren Baghdasaryan <surenb@google.com>,
	Minchan Kim <minchan@google.com>,
	Joel Fernandes <joelaf@google.com>,
	David Rientjes <rientjes@google.com>,
	Axel Rasmussen <axelrasmussen@google.com>,
	Andy Lutomirski <luto@kernel.org>,
	Michel Lespinasse <michel@lespinasse.org>
Subject: [PATCH 01/35] [PATCH v2 01/35] mm: export dump_mm
Date: Fri, 28 Jan 2022 05:09:32 -0800
Message-Id: <20220128131006.67712-2-michel@lespinasse.org>
X-Mailer: git-send-email 2.20.1
In-Reply-To: <20220128131006.67712-1-michel@lespinasse.org>
References: <20220128131006.67712-1-michel@lespinasse.org>
MIME-Version: 1.0
X-Rspamd-Server: rspam06
X-Rspamd-Queue-Id: 2D869180002
X-Stat-Signature: 5i4gwkjg7doyojxdjsk8iiw1fgtqz66m
Authentication-Results: imf16.hostedemail.com;
	dkim=pass header.d=lespinasse.org header.s=srv-52-ed header.b=M7pzfh5Z;
	dkim=pass header.d=lespinasse.org header.s=srv-52-rsa header.b=hH+5ulWn;
	dmarc=pass (policy=none) header.from=lespinasse.org;
	spf=pass (imf16.hostedemail.com: domain of walken@lespinasse.org designates
 63.205.204.226 as permitted sender) smtp.mailfrom=walken@lespinasse.org
X-Rspam-User: nil
X-HE-Tag: 1643375409-825778
X-Bogosity: Ham, tests=bogofilter, spamicity=0.000000, version=1.2.4
Sender: owner-linux-mm@kvack.org
Precedence: bulk
X-Loop: owner-majordomo@kvack.org
List-ID: <linux-mm.kvack.org>

This is necessary in order to allow VM_BUG_ON_MM to be used in modules
(I encountered the issue when adding VM_BUG_ON_MM in mmap locking functions).

Signed-off-by: Michel Lespinasse <michel@lespinasse.org>
---
 mm/debug.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mm/debug.c b/mm/debug.c
index bc9ac87f0e08..40d3f358b75c 100644
--- a/mm/debug.c
+++ b/mm/debug.c
@@ -220,6 +220,7 @@ void dump_mm(const struct mm_struct *mm)
 		mm->def_flags, &mm->def_flags
 	);
 }
+EXPORT_SYMBOL(dump_mm);
 
 static bool page_init_poisoning __read_mostly = true;
 

